subsys/usb/class/netusb/netusb.c

```diff
@@ -232,67 +203,82 @@ int try_write(u8_t ep, u8_t *data, u16_t len)
 	return ret;
 }
 
-static int netusb_device_init(struct device *dev)
+static void netusb_init(struct net_if *iface)
 {
-	struct device *eth_dev;
-	struct net_if *iface;
+	struct netusb_dev *net = NETDEV_DATA(iface);
 	int ret;
 
-	ARG_UNUSED(dev);
-
-	SYS_LOG_DBG("netusb device initialization");
-
-	eth_dev = device_get_binding(CONFIG_ETH_EMU_0_NAME);
-	if (!eth_dev) {
-		SYS_LOG_ERR("Cannot get Ethernet Emulation device");
-		return -ENODEV;
-	}
-
-	eth_emu_ctx = (struct eth_emu_context *)eth_dev->driver_data;
+	net->iface = iface;
 
-	iface = eth_emu_ctx->iface;
+	SYS_LOG_ERR("netusb_init");
 
-	SYS_LOG_DBG("iface %p", iface);
+	net_if_set_link_addr(iface, net->mac_addr, sizeof(net->mac_addr),
+			     NET_LINK_ETHERNET);
 
 	net_if_down(iface);
 
 #if defined(CONFIG_USB_DEVICE_NETWORK_ECM)
-	netusb.func[0] = ecm_register_function(iface,
-					       CONFIG_CDC_ECM_IN_EP_ADDR);
+	net->func = ecm_register_function();
 #endif
 
+	net->usb_config.endpoint = net->func->ep;
+	net->usb_config.num_endpoints = net->func->num_ep;
+
 #if defined(CONFIG_USB_COMPOSITE_DEVICE)
 #if defined(CONFIG_USB_DEVICE_NETWORK_ECM)
-	ret = composite_add_function(&netusb_config, FIRST_IFACE_CDC_ECM);
+	ret = composite_add_function(&net->usb_config, FIRST_IFACE_CDC_ECM);
 	if (ret < 0) {
 		SYS_LOG_ERR("Failed to add CDC ECM function");
-		return ret;
+		return;
 	}
 #endif /* CONFIG_USB_DEVICE_NETWORK_ECM */
 #else /* CONFIG_USB_COMPOSITE_DEVICE */
-	netusb_config.interface.payload_data = interface_data;
-	netusb_config.usb_device_description = usb_get_device_descriptor();
+	net->usb_config.interface.payload_data = interface_data;
+	net->usb_config.usb_device_description = usb_get_device_descriptor();
 
-	ret = usb_set_config(&netusb_config);
+	ret = usb_set_config(&net->usb_config);
 	if (ret < 0) {
 		SYS_LOG_ERR("Failed to configure USB device");
-		return ret;
+		return;
 	}
 
-	ret = usb_enable(&netusb_config);
+	ret = usb_enable(&net->usb_config);
 	if (ret < 0) {
 		SYS_LOG_ERR("Failed to enable USB");
-		return ret;
+		return;
 	}
 #endif /* CONFIG_USB_COMPOSITE_DEVICE */
 
 	SYS_LOG_INF("netusb initialized");
+}
 
-	return ret;
+static struct netusb_dev netusb0 = {
+	/* Use RFC7042 Documentation for MAC address */
```

Comment:
why you name it netusb0? are you going to somehow use netusb1, etc?

Reply:
who knows ? ;) this is to highlight that this struct is linked to the instance of the device we create.

Reply:
IMO even with multiple configurations we always have only one instance of this structure, so just name it netusb to make code consistent.